### ISSUE TITLE
Support Python 3.10

### ISSUE THREAD

**Issue by cjolowicz** [cjolowicz/cookiecutter-hypermodern-python#580]
Use [deadsnakes](https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa).

**Comment by cjolowicz** [comment]
Pending official support on GitHub's side, we can use [deadsnakes/action].

[deadsnakes/action]: https://github.com/deadsnakes/action

**Comment by cjolowicz** [comment]
3.10 is now available.

https://github.com/actions/python-versions/releases/tag/3.10.0-alpha.1-84873

**Comment by cjolowicz** [comment]
See cjolowicz/cookiecutter-hypermodern-python-instance#358

**Comment by cjolowicz** [comment]
See https://github.com/cjolowicz/cookiecutter-hypermodern-python/issues/817#issuecomment-808821863

This is blocked by missing support in GA for allowing jobs to fail.

Meanwhile, Nox has learned how to override the configured interpreters for a session. So this is pretty much solved locally. It would still be nice to have this run in CI though.
